src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -60,35 +69,88 @@ public static String generateClassBeanName(Class<?> beanClass) {
   }
 
   /**
-   * Returns a list of bean dependencies (as {@link BeanDependency}) this bean depends on,
-   * including constructor argument types, fields, and methods annotated with the
-   * {@link Autowired} annotation.
+   * Returns a list of bean constructor dependencies (as {@link BeanDependency}) this bean depends
+   * on by reading constructor argument types.
    *
-   * @param beanClass the class of the bean to scan for dependencies
-   * @return a list of bean dependency types
+   * @param beanConstructor the constructor of the bean to analyze for dependencies
+   * @return a list of bean constructor dependency types
    */
-  public static List<BeanDependency> getBeanDependencies(Class<?> beanClass) {
-    log.trace("Scanning class {} for @Autowire candidates", beanClass.getName());
-
-    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
-
-    validateBeanClassConstructorSize(beanClass.getName(), declaredConstructors);
+  public static List<BeanDependency> getConstructorBeanDependencies(
+      Constructor<?> beanConstructor) {
+    Objects.requireNonNull(beanConstructor, () -> {
+      log.error("Failed to get bean constructor dependencies for nullable constructor");
+      return "Bean constructor has not been specified";
+    });
+    log.trace("Scanning class {} for @Autowire candidates", beanConstructor.getDeclaringClass());
 
-    Stream<Class<?>> constructorDependenciesStream = Arrays.stream(declaredConstructors)
-        .flatMap(constructor -> Arrays.stream(constructor.getParameterTypes()));
+    return Arrays.stream(beanConstructor.getParameterTypes())
+        .map(dependency -> new BeanDependency(generateClassBeanName(dependency), dependency))
+        .toList();
+  }
 
-    Stream<Class<?>> fieldDependenciesStream = Arrays.stream(beanClass.getDeclaredFields())
-        .filter(field -> field.isAnnotationPresent(Autowired.class))
-        .map(Field::getType);
+  /**
+   * Finds and returns the initialization constructor for a given bean class. The initialization
+   * constructor is determined based on the presence of the {@link Autowired} annotation and the
+   * number of available constructors.
+   *
+   * @param beanClass the class of the bean to find the initialization constructor for.
+   * @param beanName  the name of the bean, used for logging and error handling.
+   * @return the initialization constructor for the specified bean class.
+   * @throws BeanDefinitionCreationException If there are issues with the bean's constructors, such
+   *                                         as having more than one constructor without the
+   *                                         {@link Autowired} annotation or multiple constructors
+   *                                         marked with {@link Autowired}.
+   */
+  public static Constructor<?> findBeanInitConstructor(Class<?> beanClass, String beanName) {
```

Comment:
I think beanName is redundant, it does not give us additional information, because constructors are related only to the class implementations, so if we have an issue we have an issue with the class

Reply:
From my point of view we can leave it to make error handling a little bit more user friendly.